waspc/data/Generator/templates/react-app/src/queryClient.js

```diff
@@ -1,3 +1,21 @@
 import { QueryClient } from 'react-query'
 
-export const queryClient = new QueryClient()
+let queryClientConfig, resolveQueryClientInitialized, isQueryClientInitialized
+
+export const queryClientInitialized = new Promise(resolve => {
+  resolveQueryClientInitialized = resolve
+});
+
+export function configureQueryClient(config) {
+  if (isQueryClientInitialized) {
+    throw new Error("Attempted to configure the QueryClient after initialization")
+  }
+
+  queryClientConfig = config
+}
+
+export function initializeQueryClient() {
+  const queryClient = new QueryClient(queryClientConfig)
```

Comment:
So what happens then if they didn't set the queryClientConfig? Its value is undefined, and new QueryClient with default settings will be created?
Should we explicitly set queryClientConfig to {} just to make this clear, or maybe do `new QueryClient(queryClientConfig || defaultQueryClientConfig)` here, where we previously define `defaultQueryClientConfig` as `{}` or whatever we want it to be?

Reply:
I'll do the default thing, it's cleaner. This relies a bit too much on knowing what react query does under the hood.